src/Client/SidebarComponents/Navbar.fs

```diff
@@ -1,25 +1,374 @@
+                                }
+                            | investigationIdentifier when investigationIdentifier.ToLower().Contains("investigation") ->
+                                let! investigation = OfficeInterop.Core.tryGetTopLeveMetadata (investigationIdentifier.ToLower()) ArcInvestigation.fromMetadataCollection
+                                setExcelMetadataType {
+                                    excelMetadataType with
+                                        MetadataType = Some ArcFilesDiscriminate.Investigation
+                                        Identifier = Some investigationIdentifier
+                                        Investigation = if investigation.IsSome then investigation else Some (new ArcInvestigation("New Investigation"))
+                                }
+                            | studyIdentifier when studyIdentifier.ToLower().Contains("study") ->
+                                let! study = OfficeInterop.Core.tryGetTopLeveMetadata (studyIdentifier.ToLower()) ArcStudy.fromMetadataCollection
+                                setExcelMetadataType {
+                                    excelMetadataType with
+                                        MetadataType = Some ArcFilesDiscriminate.Study
+                                        Identifier = Some studyIdentifier
+                                        Study = if study.IsSome then study else Some (new ArcStudy("New Study"), [])
+                                }
+                            | templateIdentifier when templateIdentifier.ToLower().Contains("template") ->
+                                let! topLevelTemplateInfo = OfficeInterop.Core.tryGetTopLeveMetadata (templateIdentifier.ToLower()) Template.fromMetadataCollection
+                                let template =
+                                    if topLevelTemplateInfo.IsSome then 
+                                        let templateInfo, ers, tags, authors = topLevelTemplateInfo.Value                                
+                                        Some (Template.fromParts templateInfo ers tags authors (ArcTable.init "New Template") DateTime.Now)
+                                    else Some (new Template(Guid.NewGuid(), (ArcTable.init "New Template")))
+                                setExcelMetadataType {
+                                    excelMetadataType with
+                                        MetadataType = Some ArcFilesDiscriminate.Template
+                                        Identifier = Some templateIdentifier
+                                        Template = template
+                                }
+                            | _ -> failwith $"No metadata of type {result.Value} has been implemented yet!"
+                        else setExcelMetadataType(ExcelMetadataState.init)
+                    }
+                )  |> ignore
+            )
```

Comment:
can show a gif of this ui?

i can recommend screentogif as tool

Reply:
![ExcelSwateTopLevelMetadataShowcase18102024PB](https://github.com/user-attachments/assets/7d96092e-4f39-4135-9fdc-46bfc6c45ad7)
thanks for the tool suggestion! It is really cool, I hope this is helpful